.github/workflows/ci.yml

```diff
@@ -33,4 +39,26 @@ jobs:
     - name: install dependencies
       run: npm ci
     - name: test
-      run: npm run test
\ No newline at end of file
+      run: npm run test
+  release:
+      name: Release
+      runs-on: ubuntu-latest
+      needs:
+        - test
+
+      if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
Since this is a step we only want to trigger when a tag is pushed I would create its own GitHub Workflow, this way we don't see this in PRs:

![image](https://user-images.githubusercontent.com/2574275/189429493-f1500952-4e22-4581-94b3-b104d41f3e1a.png)

it doesn't add much value.

Reply:
So the reason it is in one file is i would like it to make sure the test pass before it publishes, im not sure i can do that if they are separate files. Do you know if that is possible?

Reply:
You can use `on.workflow_run` and filter by branch: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#limiting-your-workflow-to-run-based-on-branches

Reply:
So i couldn't find a way to only work with pushes on tag so i have just made it publish to the registry as that is what we seems to do for other packages.